### PR TITLE
Replace channel slash transaction with extended solo close transaction

### PR DESCRIPTION
**Pull request by gorillainduction**
The difference in semantics of the two transactions were only when
they could be used. Now the solo close transaction can be used as
a slash transaction instead.

https://www.pivotaltracker.com/n/projects/2124891/stories/158421035

### CODE REVIEW

File: apps/aechannel/src/aesc_close_solo_tx.erl [hunk 1]
```diff
@@ -216,28 +216,45 @@ serialization_template(?CHANNEL_CLOSE_SOLO_TX_VSN) ->
...
+    InitiatorPubKey   = aesc_channels:initiator(Channel),
+    ResponderPubKey   = aesc_channels:responder(Channel),
+    ChannelAmount     = aesc_channels:total_amount(Channel),
+    case aesc_utils:accounts_in_poi([InitiatorPubKey, ResponderPubKey], PoI) of
+        {error, _} = Err -> Err;
+        {ok, [PoIInitiatorAcc, PoIResponderAcc]} ->
+            PoIInitiatorAmt = aec_accounts:balance(PoIInitiatorAcc),
+            PoIResponderAmt = aec_accounts:balance(PoIResponderAcc),
+            PoIAmount       = PoIInitiatorAmt + PoIResponderAmt,
+            case ChannelAmount =:= PoIAmount of
```
Comment (uwiger) [inline apps/aechannel/src/aesc_close_solo_tx.erl]: Would it not be acceptable here with `ChannelAmount >= PoIAmount`?

Comment (gorillainduction) [reply apps/aechannel/src/aesc_close_solo_tx.erl]: Yes. This is a restriction that we have discussed to lift, but I aimed at not changing the semantics for now.

**Comment by lucafavatella** [comment]
I took the liberty of labelling as WIP for the time being - it eases filtering.

**Comment by gorillainduction** [comment]
Closing this PR because of the reasons @lucafavatella gives above.

**Pull request closed without merging**